packages/gitbook/src/components/TableOfContents/useScrollToActiveTOCItem.tsx

```diff
@@ -0,0 +1,84 @@
+'use client';
+
+import React from 'react';
+
+import { tcls } from '@/lib/tailwind';
+
+const TOCScrollContainerContext = React.createContext<React.RefObject<HTMLDivElement> | null>(null);
+
+export function TOCScrollContainerProvider(
+    props: React.PropsWithChildren<{ withHeaderOffset: boolean; withTrademarkEnabled: boolean }>,
+) {
+    const { withHeaderOffset, withTrademarkEnabled, children } = props;
+    const scrollContainerRef = React.createRef<HTMLDivElement>();
+
+    return (
+        <TOCScrollContainerContext.Provider value={scrollContainerRef}>
+            <div
+                ref={scrollContainerRef}
+                className={tcls(
```

Comment:
Instead of passing the props here and computing the `className` here, you can pass the className as a prop or even accept `props: React.HTMLAttributes<HTMLDivElement>` and pass them all to the div